md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/ext/HiveTableMigrationAPI.java

```diff
@@ -247,4 +259,115 @@ public RestWrapper createJob(@RequestParam Map<String, String> map, Principal pr
 
     }
 
+
+    @RequestMapping(value = {"/cluster/{description}"}, method = RequestMethod.GET)
+    @ResponseBody
+    public RestWrapper list(@PathVariable("description") String description, Principal principal) {
+
+        RestWrapper restWrapper = null;
+        try {
+
+            GetGeneralConfig generalConfigs = new GetGeneralConfig();
+            List<GeneralConfig> generalConfigList = generalConfigs.byLikeConfigGroup(description, 1);
+            if (!generalConfigList.isEmpty()) {
+                if (generalConfigList.get(0).getRequired() == 2) {
+                    restWrapper = new RestWrapper("Listing of Records Failed", RestWrapper.ERROR);
+                } else {
+                    restWrapper = new RestWrapper(generalConfigList, RestWrapper.OK);
+                    LOGGER.info("All records listed with config group :" + "cluster" + "from General  Config by User:" + principal.getName());
+                }
+            } else {
+                restWrapper = new RestWrapper(generalConfigList, RestWrapper.OK);
+
+                LOGGER.info("All records listed with config group :" + "cluster" + "from General  Config by User:" + principal.getName());
+            }
+
+        } catch (MetadataException e) {
+            LOGGER.error(e);
+            restWrapper = new RestWrapper("Exception while fetching details of cluster", RestWrapper.ERROR);
+        }
+        return restWrapper;
+    }
+
+    @RequestMapping(value = {"/insertcluster", "insertcluster"}, method = RequestMethod.PUT)
+
+    @ResponseBody
+    public RestWrapper insertCluster(@ModelAttribute("clusterInfo")
+                                     @Valid ClusterInfo cluster, BindingResult bindingResult, Principal principal) {
+
+        RestWrapper restWrapper = null;
+        if (bindingResult.hasErrors()) {
+            BindingResultError bindingResultError = new BindingResultError();
+            return bindingResultError.errorMessage(bindingResult);
+        }
+        try {
+            LOGGER.info("name_node_host " + cluster.getNameNodeHostName() + ":" + cluster.getNameNodePort());
+            generalConfigDAO.insertCluster(cluster);
+
+
+            restWrapper = new RestWrapper(cluster, RestWrapper.OK);
+            LOGGER.info("Record inserted in General Config by User:" + principal.getName());
+
+        } catch (MetadataException e) {
+            LOGGER.error(e);
+            restWrapper = new RestWrapper(e.getMessage(), RestWrapper.ERROR);
+        }
+        return restWrapper;
+    }
+
+    @RequestMapping(value = {"/updatecluster", "updatecluster"}, method = RequestMethod.POST)
+
+    @ResponseBody
+    public RestWrapper updateCluster(@RequestParam Map<String, String> map, Principal principal) {
+
+        RestWrapper restWrapper = null;
+        String description;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "description" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1481)